### PR TITLE
client: fix bundle index

### LINKED ISSUE #16: Transfer doesn't create a bundle with multiple transactions
If I send a transaction with a long message that doesn't fit into a single signatureMessageFragement, a second transaction is created with the same bundle hash, but at the same index 0 instead of 1:
https://thetangle.org/bundle/CGGYHCWAABKSHHBIQYIIOOHKKKQSFBXRJOEKEXGNVVHVLEHIIUKPBBSHFNSHKCPLEAKGZEVKZHHHPCGPZ

### PR DESCRIPTION
**Pull request by wusyong**
Fix #16

### CODE REVIEW

File: iota-model/bundle.rs [hunk 1]
```diff
@@ -101,6 +101,20 @@ impl Bundle {
         }
     }
 
+    /// Reset bundle indexes
+    pub fn reset_indexes(&mut self) -> Result<()> {
```
Comment (njaremko) [inline iota-model/bundle.rs]: Nothing in here returns an error, so why are you returning a result?

**Comment by njaremko** [comment]
You shouldn't amend commits and force push in a PR, you mess with the historical record. You can choose to squash when the PR is merged

**Comment by wusyong** [comment]
I think it's just preference. Personally if the change is trivial, I prefer to make commit log identical to the subject.

**Review by tsvisabo — APPROVED** [review]

**Comment by oopsmonk** [comment]
> I think it's just preference. Personally if the change is trivial, I prefer to make commit log identical to the subject.

I agree with @njaremko, you should use `squash and merge`.

**Pull request merged**